### ISSUE TITLE
Documentation of installation process

### ISSUE THREAD

**Issue by RHodzelmans** [openshift/installer#2328]
# Version
openshift-install-mac-4.2.0-0.nightly-2019-09-05-234433

```console
$ openshift-install version
./openshift-install v4.2.0
built from commit 8c9abe40f7616303c03cafdc9ad612cd8fa7bd6b
release image quay.io/openshift-release-dev/ocp-release-nightly@sha256:1c31f16cdec4095d13a5e11e9b3a62282c26b23673deb6bb0c9cb214e2da850a
```

# Platform:
Azure

# What happened?
Documentation is unclear.

My issue is with: 
- https://github.com/openshift/installer/blob/master/docs/user/azure/credentials.md
- https://github.com/openshift/installer/blob/master/docs/user/azure/install.md

Please split the former into two separate documentations: one for using the CLI - with commands (as is today) and one for using the GUI - using screenshots. Then the remarks:
- What if no user shows up at `az ad sp list --show-mine -otable` but it does exist? Is that an issue?
- Step 2 does not explain or link to a GUI equivalent at all.
- What is the API and the permission we're granting in?
`az ad app permission add --id <AppId> --api 00000002-0000-0000-c000-000000000000 --api-permissions 824c81eb-e3f8-4ee6-8f6d-de7f50d565b7=Role`
- How to grant that permission in the GUI?
- Who is team-installer and where is it coming from? 
`$ az ad sp create-for-rbac --role Owner --name team-installer | jq --arg sub_id "$(az account show | jq -r '.id')" '{subscriptionId:$sub_id,clientId:.appId, clientSecret:.password,tenantId:.tenant}' > ~/.azure/osServicePrincipal.json`

For the latter:

```[~]$ openshift-install create install-config
? SSH Public Key /home/user_id/.ssh/id_rsa.pub
? Platform azure
? azure subscription id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
? azure tenant id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
? azure service principal client id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
? azure service principal client secret xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
? Region centralus
? Base Domain example.com
? Cluster Name test
? Pull Secret [? for help
```

This dialog only is available the first run. Second run I'm getting:

```[~]$ openshift-install create install-config
? SSH Public Key /home/user_id/.ssh/id_rsa.pub
? Platform azure
? Region centralus
? Base Domain example.com
? Cluster Name test
? Pull Secret [? for help
```

Where are the azure prefixed lines cached / stored, because OCP install does not ask the next run? Or are these properties I can edit via commands? If so, which? I want to verify these during troubleshooting. Eg I created a new Service Principle, yet there seems no option to modify those settings in the installer. 

# Expectation
Please, can the documentation to be updated? It would help.

**Comment by abhinavdahiya** [comment]
> My issue is with:
> 
>     * https://github.com/openshift/installer/blob/master/docs/user/azure/credentials.md
> 
>     * https://github.com/openshift/installer/blob/master/docs/user/azure/install.md
> 
> 
> Please split the former into two separate documentations: one for using the CLI - with commands (as is today) and one for using the GUI - using screenshots. Then the remarks:
> 
>     * What if no user shows up at `az ad sp list --show-mine -otable` but it does exist? Is that an issue?

If the service principal was created by you as mentioned in step 1, the service principal should show up. 

> 
>     * Step 2 does not explain or link to a GUI equivalent at all.

The step links you to https://docs.microsoft.com/en-us/azure/active-directory/develop/v1-permissions-and-consent to understand AD permissions and how to get them. There is no requirement for us to document them again, as Azure docs should be the source of truth.

The cli is provided for ease.

>     * What is the API and the permission we're granting in?
>       `az ad app permission add --id <AppId> --api 00000002-0000-0000-c000-000000000000 --api-permissions 824c81eb-e3f8-4ee6-8f6d-de7f50d565b7=Role`

https://github.com/openshift/installer/blame/master/docs/user/azure/credentials.md#L14 provides the details on what permission we need for the service principal.

>     * Who is team-installer and where is it coming from?
>       `$ az ad sp create-for-rbac --role Owner --name team-installer | jq --arg sub_id "$(az account show | jq -r '.id')" '{subscriptionId:$sub_id,clientId:.appId, clientSecret:.password,tenantId:.tenant}' > ~/.azure/osServicePrincipal.json`

that is just an example of how you can create an service principal for the subscription and store in a location that is accessible by installer.. so I don't get what you expect here?
 
> 
> For the latter:
> 
> ```
> ? SSH Public Key /home/user_id/.ssh/id_rsa.pub
> ? Platform azure
> ? azure subscription id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
> ? azure tenant id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
> ? azure service principal client id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
> ? azure service principal client secret xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
> ? Region centralus
> ? Base Domain example.com
> ? Cluster Name test
> ? Pull Secret [? for help
> ```
> 
> This dialog only is available the first run. Second run I'm getting:
> 
> ```
> ? SSH Public Key /home/user_id/.ssh/id_rsa.pub
> ? Platform azure
> ? Region centralus
> ? Base Domain example.com
> ? Cluster Name test
> ? Pull Secret [? for help
> ```
> 
> Where are the azure prefixed lines cached / stored, because OCP install does not ask the next run? Or are these properties I can edit via commands? If so, which? I want to verify these during troubleshooting. Eg I created a new Service Principle, yet there seems no option to modify those settings in the installer.

https://github.com/openshift/installer/blame/master/docs/user/azure/credentials.md#L49 already provides you the information about the default and how to provide explicit.. 

> # Expectation
> 
> Please, can the documentation to be updated? It would help.

All your issues seem don't seem like they would require a change in documentation as they are already covered.. but still we are open to PRs with changes that can make our documentation better :) 

https://github.com/openshift/installer/blob/master/CONTRIBUTING.md

**Comment by RHodzelmans** [comment]
> > ```
> > * What if no user shows up at `az ad sp list --show-mine -otable` but it does exist? Is that an issue?
> > ```
> 
> If the service principal was created by you as mentioned in step 1, the service principal should show up.

It doesn't - the service principle exists in my account, I can query it and it exists, but as soon as I add --show-mine it's not showing up.

> 
> > ```
> > * Step 2 does not explain or link to a GUI equivalent at all.
> > ```
> 
> The step links you to https://docs.microsoft.com/en-us/azure/active-directory/develop/v1-permissions-and-consent to understand AD permissions and how to get them. There is no requirement for us to document them again, as Azure docs should be the source of truth.
> 
> The cli is provided for ease.
The UI is easier ;-)
> 
> > ```
> > * What is the API and the permission we're granting in?
> >   `az ad app permission add --id <AppId> --api 00000002-0000-0000-c000-000000000000 --api-permissions 824c81eb-e3f8-4ee6-8f6d-de7f50d565b7=Role`
> > ```
> 
> https://github.com/openshift/installer/blame/master/docs/user/azure/credentials.md#L14 provides the details on what permission we need for the service principal.
Makes sense - I must have missed that.

> 
> > ```
> > * Who is team-installer and where is it coming from?
> >   `$ az ad sp create-for-rbac --role Owner --name team-installer | jq --arg sub_id "$(az account show | jq -r '.id')" '{subscriptionId:$sub_id,clientId:.appId, clientSecret:.password,tenantId:.tenant}' > ~/.azure/osServicePrincipal.json`
> > ```
> 
> that is just an example of how you can create an service principal for the subscription and store in a location that is accessible by installer.. so I don't get what you expect here?

so Owner is the AD role, team-installer is my self chosen name? I'd use `<your service principe name> `
> 
> > For the latter:
> > ```
> > ? SSH Public Key /home/user_id/.ssh/id_rsa.pub
> > ? Platform azure
> > ? azure subscription id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
> > ? azure tenant id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
> > ? azure service principal client id xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
> > ? azure service principal client secret xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
> > ? Region centralus
> > ? Base Domain example.com
> > ? Cluster Name test
> > ? Pull Secret [? for help
> > ```
> > 
> > 
> > This dialog only is available the first run. Second run I'm getting:
> > ```
> > ? SSH Public Key /home/user_id/.ssh/id_rsa.pub
> > ? Platform azure
> > ? Region centralus
> > ? Base Domain example.com
> > ? Cluster Name test
> > ? Pull Secret [? for help
> > ```
> > 
> > 
> > Where are the azure prefixed lines cached / stored, because OCP install does not ask the next run? Or are these properties I can edit via commands? If so, which? I want to verify these during troubleshooting. Eg I created a new Service Principle, yet there seems no option to modify those settings in the installer.
> 
> https://github.com/openshift/installer/blame/master/docs/user/azure/credentials.md#L49 already provides you the information about the default and how to provide explicit..

Ok, so those properties are cached there. It does not repeat that on this page:
https://github.com/openshift/installer/blob/master/docs/user/azure/install.md

I'd expect documentation to refer back. 

> 
> > # Expectation
> > Please, can the documentation to be updated? It would help.
> 
> All your issues seem don't seem like they would require a change in documentation as they are already covered.. but still we are open to PRs with changes that can make our documentation better :)

Let me try again, but 
> 
> https://github.com/openshift/installer/blob/master/CONTRIBUTING.md

**Comment by RHodzelmans** [comment]
I was hoping something along the lines of this: https://blog.openshift.com/openshift-4-2-on-azure-preview/

**Comment by abhinavdahiya** [comment]
I think the documentation at https://github.com/openshift/installer/tree/master/docs/user/azure provides steps to follow to create a cluster on azure.

I answered most of the concerns https://github.com/openshift/installer/issues/2328#issuecomment-529573960
if you have precise areas for improvement on the docs please open a separate issue for that.

/close
